src/crypto/CHIPCryptoPAL.h

```diff
@@ -472,6 +488,13 @@ using P256SerializedKeypair = SensitiveDataBuffer<kP256_PublicKey_Length + kP256
 
 class P256KeypairBase : public ECPKeypair<P256PublicKey, P256ECDHDerivedSecret, P256ECDSASignature>
 {
+protected:
+    // This base type can't be copied / assigned directly.
```

Comment:
quick question, how does this prevent subtypes from either not being copyable or final?

Or is this just a general comment that ends up being enforced by code audits?

Reply:
@tehampson 
^ Just mentioning myself so that I get emails when reply happens on this PR

Reply:
Having them protected means you can't directly use the base type to do a copy, e.g.
```P256KeypairBase copy = someOtherKaypair;```

Each sub-class still has to deal with whether copying should be possible or not.

Reply:
> Or is this just a general comment that ends up being enforced by code audits?

Yes, that's right.

Reply:
> Having them protected means you can't directly use the base type to do a copy, e.g.
`P256KeypairBase copy = someOtherKaypair;`
Each sub-class still has to deal with whether copying should be possible or not.

Thanks for that explanation, very clear now.